File: JetpacReloaded/JetpacReloaded/Particle.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;

namespace JetpacReloaded
{
    public class Particle : BaseObject
    {
        private readonly int lifeSpan = 20;
        private int particleX;
        private int particleY;
        private int length = 0;
        private int count = 0;
        private float scale = 1.0f;
        private readonly Color[] colors = {Color.White,
                          Color.Yellow,
                          Color.Orange,
                          Color.Brown,
                          Color.Red,
                          Color.Brown,
                          Color.Brown,
                          Color.Black };

        public Particle(Texture2D image)
        {
            m_image = image;
        }

        public void Update(int x, int y, bool facingLeft, bool showParticles)
        {
            if (showParticles == true)
            {
                if (particleY < length)
                {
                    particleY++;
                }
                if (particleY >= length)
                {
                    if (facingLeft == true)
                    {
                        particleX = x + 20 + rand.Next(0, 10);
                        length = y + 40 + lifeSpan;
                        particleY = y + 40 + rand.Next(0, 10);
                    }
                    else
                    {
                        particleX = x + rand.Next(0, 10);
                        length = y + 40 + lifeSpan;
                        particleY = y + 40 + rand.Next(0, 10);
                    }
                }

                if (particleY == (length - 20))
                {
                    count = 0;
                    scale = 1.0f;
                }
                if (particleY == (length - 15))
                {
                    count = 1;
                    scale = 0.9f;
                }
                if (particleY == (length - 10))
                {
                    count = 2;
                    scale = 0.8f;
                }
                if (particleY == (length - 5))
                {
                    count = 3;
                    scale = 0.7f;
                }
                if (particleY == (length))
                {
                    count = 4;
                    scale = 0.45f;
                }
            }

            // Hide the particles when not in use
            if (showParticles == false)
            {
                count = 7;
                particleX = 0;
                particleY = 0;
                length = 0;
            }
        }

        new public void Draw(SpriteBatch spriteBatch, Camera camera)
        {
            spriteBatch.Begin(SpriteSortMode.Immediate, BlendState.Additive, transformMatrix: camera.Transform);
            Vector2 particleLocation = new Vector2(particleX, particleY);
            spriteBatch.Draw(m_image, particleLocation, null, colors[count], 0, new Vector2(0, 9), scale, SpriteEffects.None, 0);
            spriteBatch.End();
        }
    }
}

File: JetpacReloaded/Typescript version/HTML5Canvas/ts/particle.ts
import { Color } from "../ts_lib/color";
import { SpriteBatch, Texture2D } from "../ts_lib/spritebatch";
import { Vector2 } from "../ts_lib/vector2.js";
import { BaseObject } from "./baseobject.js";
import { Camera } from "./camera.js";

export class Particle extends BaseObject {
    private lifeSpan: number = 20;
    private particleX: number;
    private particleY: number;
    private length: number = 0;
    private count: number = 0;
    private scale: number = 1.0;
    private colors: Color[] = [Color.White,
    Color.Yellow,
    Color.Orange,
    Color.Brown,
    Color.Red,
    Color.Brown,
    Color.Brown,
    Color.Black];

    constructor(image: Texture2D) {
        super();
        this.m_image = image;
    }
    
    public Update(x: number, y: number, facingLeft: boolean, showParticles: boolean): void {
        if (showParticles == true) {
            if (this.particleY < this.length) {
                this.particleY++;
            }
            if (this.particleY >= this.length) {
                if (facingLeft == true) {
                    this.particleX = x + 20 + this.rand.Next(0, 10);
                    this.length = y + 40 + this.lifeSpan;
                    this.particleY = y + 40 + this.rand.Next(0, 10);
                }
                else {
                    this.particleX = x + this.rand.Next(0, 10);
                    this.length = y + 40 + this.lifeSpan;
                    this.particleY = y + 40 + this.rand.Next(0, 10);
                }
            }
            if (this.particleY == (this.length - 20)) {
                this.count = 0;
                this.scale = 1.0;
            }
            if (this.particleY == (this.length - 15)) {
                this.count = 1;
                this.scale = 0.9;
            }
            if (this.particleY == (this.length - 10)) {
                this.count = 2;
                this.scale = 0.8;
            }
            if (this.particleY == (this.length - 5)) {
                this.count = 3;
                this.scale = 0.7;
            }
            if (this.particleY == (this.length)) {
                this.count = 4;
                this.scale = 0.45;
            }
        }
        if (showParticles == false) {
            this.count = 7;
            this.particleX = 0;
            this.particleY = 0;
            this.length = 0;
        }
    }
    // public Draw(spriteBatch: SpriteBatch, camera: Camera): void {
    //     spriteBatch.Begin(SpriteSortMode.Immediate, BlendState.Additive,/*transformMatrix:*/camera.Transform);
    //     let particleLocation: Vector2 = new Vector2(this.particleX, this.particleY);
    //     spriteBatch.Draw(this.m_image, particleLocation, null, this.colors[this.count], 0, new Vector2(0, 9), this.scale, SpriteEffects.None, 0);
    //     spriteBatch.End();
    // }
}